marlowe-cardano-minimal/app/Benchmark/Marlowe/RolePayout.hs

```diff
@@ -0,0 +1,146 @@
+
+-----------------------------------------------------------------------------
+--
+-- Module      :  $Headers
+-- License     :  Apache 2.0
+--
+-- Stability   :  Experimental
+-- Portability :  Portable
+--
+-- | Benchmarking support for Marlowe's role-payout validator.
+--
+-----------------------------------------------------------------------------
+
+
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE RecordWildCards     #-}
+
+
+module Benchmark.Marlowe.RolePayout (
+  -- * Benchmarking
+  benchmarks
+, validatorBytes
+, validatorHash
+, exampleBenchmark
+) where
+
+
+import Benchmark.Marlowe (readBenchmarks)
+import Benchmark.Marlowe.Types (Benchmark (..), makeBenchmark)
+import Benchmark.Marlowe.Util (lovelace, makeBuiltinData, makeDatumMap, makeInput, makeOutput,
+                               makeRedeemerMap, updateScriptHash)
+import Data.Bifunctor (second)
+import Language.Marlowe.Scripts (rolePayoutValidatorBytes, rolePayoutValidatorHash)
+import PlutusLedgerApi.V2 (Credential (PubKeyCredential, ScriptCredential), ExBudget (ExBudget),
+                           Extended (NegInf, PosInf), Interval (Interval), LowerBound (LowerBound),
+                           ScriptContext (ScriptContext, scriptContextPurpose, scriptContextTxInfo),
+                           ScriptHash, ScriptPurpose (Spending), SerialisedScript, TxInfo (..),
+                           TxOutRef (TxOutRef), UpperBound (UpperBound), singleton)
+
+import PlutusTx.AssocMap qualified as AM (empty)
+
+
+-- | The serialised Marlowe role-payout validator.
+validatorBytes :: SerialisedScript
+validatorBytes = rolePayoutValidatorBytes
+
+
+-- | The hash for the Marlowe role-payout validator.
+validatorHash :: ScriptHash
+validatorHash = rolePayoutValidatorHash
+
+
+-- | The benchmark cases for the Marlowe role-payout validator.
+benchmarks :: IO (Either String [Benchmark])
+benchmarks =
+  let
+    update benchmark@Benchmark{..} =
+      benchmark {
+        bScriptContext =
+          updateScriptHash
+            "e165610232235bbbbeff5b998b233daae42979dec92a6722d9cda989"
+            rolePayoutValidatorHash
+            bScriptContext
+      }
+  in
+    second (update <$>) <$> readBenchmarks "rolepayout"
+
+
+-- | An example benchmark for the Marlowe role-payout validator.
```

Comment:
what's this for? debugging?

Reply:
Generally it seems like we have a bunch of stuff that just supports the creation of these, whereas the "real" benchmarks we're just reading them from files?